README.md

```diff
@@ -34,38 +39,35 @@ const WebGreeting = reactToWebComponent(Greeting, React, ReactDOM);
 customElements.define("web-greeting", WebGreeting);
 ```
 
+Now we can use `<web-greeting>` like any other HTML element!
 
-Now you can use `<web-greeting>` like any other HTML element!
-
-You can create it programatically:
-
-```js
-const webGreeting = document.createElement("web-greeting");
-webGreeting.name = "StandardsFan";
-
-document.body.append(webGreeting);
+```html
+<body>
+  <h1>Greeting Demo</h1>
 
-webGreeting.innerHTML //-> "<h1>Hello, StandardsFan</h1>"
+  <web-greeting></web-greeting>
+</body>
 ```
 
-Or you can use it declaratively:
+Note that by using React 18, `reactToWebComponent` will use the new root API. If your application needs the legacy API, please use React 17 
 
-```js
-document.body.innerHTML = "<web-greeting></web-greeting>";
-
-document.body.firstChild.name = "CoolBeans";
 
-document.body.firstChild.innerHTML //-> "<h1>Hello, CoolBeans</h1>"
-```
+In the above case, the web-greeting custom element is not making use of the ```name``` property from our ```Greeting``` component. 
 
-### Working with Attributes
+## Working with Attributes
 
 By default, custom elements created by `reactToWebComponent` only
 pass properties to the underlying React component. To make attributes
 work, you must specify your component's properties with
 [PropTypes](https://reactjs.org/docs/typechecking-with-proptypes.html) as follows:
 
 ```js
+import React from 'react';
+import PropTypes from 'prop-types';
+import * as ReactDOM from 'react-dom/client';
+// When using React 16 and 17 import ReactDom with the commented statement below instead:
```

Comment:
I think we can limit this extra info to just the basic usage and assume React 18 in every other example.